### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by maniackcrudelis**
- Restart tests when a temporary error happens, to avoid complete failure for stupid errors.
- Do not fails to perform test if no install have been done before. (Allow all tests to run without a previous install test)
- Create snapshot for successful install even out of the first installations.
- Add an option `--show-resources` to show unavailable resources when accessing the url.
- Fix an error with multiple upgrade_from with some set to 0
- [Work in progress] Remove swap file before creating snapshots to prevent a crash of the CI by lack of free space.

### CODE REVIEW

**Comment by kay0u** [comment]
You have indentation problems, I think.

**Comment by maniackcrudelis** [comment]
That's because I turn my text editor to tabulation as 4 spaces instead of a tabulation.

File: sub_scripts/testing_process.sh [hunk 1]
```diff
@@ -478,16 +521,38 @@ check_test_result_remove () {
...
-		# If subdir installation worked.
-		echo subdir
-	elif [ $RESULT_check_root -eq 1 ] || [ $force_install_ok -eq 1 ]
-	then
-		# If root installation worked, return root or force_install_ok setted, return root.
-		echo root
-	else
-		ECHO_FORMAT "All installs failed, impossible to perform this test...\n" "red" clog
+    # If the test for install in sub dir isn't desactivated
+    if [ $setup_sub_dir -ne 0 ]
```
Comment (kay0u) [inline sub_scripts/testing_process.sh]:
```suggestion
    sub_dir_install=0
    if [ $setup_sub_dir -ne 0 ]
```

File: sub_scripts/testing_process.sh [hunk 2]
```diff
@@ -478,16 +521,38 @@ check_test_result_remove () {
...
+            # Validate installation in sub dir.
+            sub_dir_install=1
+        fi
+    else
+        sub_dir_install=0
+    fi
+
+    # If the test for install on root isn't desactivated
+
+    if [ $setup_root -ne 0 ] || [ $setup_nourl -eq 1 ]
```
Comment (kay0u) [inline sub_scripts/testing_process.sh]:
```suggestion
    root_install=0
    if [ $setup_root -ne 0 ] || [ $setup_nourl -eq 1 ]
```

**Comment by kay0u** [comment]
Merging time!

**Pull request merged**